ocaml/xapi/xapi_vm_appliance.ml

```diff
@@ -127,11 +123,15 @@ let assert_can_be_recovered ~__context ~self ~session_to =
     vms
 
 let get_SRs_required_for_recovery ~__context ~self ~session_to =
-  let vms = Db.VM_appliance.get_VMs ~__context ~self in
-  let sr_list = List.map
-      (fun vm -> Xapi_vm_helpers.get_SRs_required_for_recovery ~__context ~self:vm ~session_to)
-      vms in
-  List.setify(List.flatten sr_list)
+  Db.VM_appliance.get_VMs ~__context ~self
+  |> List.to_seq
+  |> Seq.map (fun vm ->
+      Xapi_vm_helpers.get_SRs_required_for_recovery ~__context ~self:vm ~session_to
+      |> List.to_seq)
+  |> Seq.flat_map Fun.id
+  |> Xapi_vm_helpers.SRSet.of_seq
+  |> Xapi_vm_helpers.SRSet.to_seq
```

Comment:
You can use `Xapi_vm_helpers.SRSet.elements` to get a list directly, instead of converting to a seq and then to a list